packages/create-sitecore-jss/src/common/processes/transform.ts

```diff
@@ -205,9 +202,15 @@ export const transform = async (
 
   for (const file of files) {
     try {
-      const pathToNewFile = `${destinationPath}\\${file}`;
+      let pathToNewFile = `${destinationPath}\\${file}`;
       const pathToTemplate = path.join(templatePath, file);
 
+      // Rename gitignore after the fact to prevent npm from renaming it to .npmignore
+      // See: https://github.com/npm/npm/issues/1862
+      if (file.endsWith('gitignore')) {
+        pathToNewFile = pathToNewFile.replace(/\gitignore$/, '.gitignore');
+      }
```

Comment:
Should we account for `file` being ".gitignore"? It would rename it as "..gitignore" in this case. Shouldn't be a problem for the published package (since .gitignore would be removed), but could encounter with template development.

Reply:
You are right, I added condition